### ISSUE TITLE
[4.0.0-rc.2] Ionicons do not show on a physical device

### ISSUE THREAD

**Issue by wcmhc** [ionic-team/ionic#17124]
# Bug Report

**Ionic version:**

[x] 4.0.0-rc.2

**Current behavior:**
Icons are not appearing on a physical android device (Android 6.0) but appear correctly on an emulator (Android 5.x)
**Expected behavior:**
Icons show on both physical and emulated devices

**Steps to reproduce:**
1. Create a basic Cordova application with dependencies listed below
2. Add a basic Ionic tab example such as the one found at the [beta tabs api example](https://beta.ionicframework.com/docs/api/tabs/)
3. Launch on a physical device (i.e. `cordova run --device`)

**Related code:**
_I have added a repo on github here: https://github.com/wcmhc/testIonicons-v4_

* index.html:
```html
<html>
   { ... }
        <link href="ionic/core/css/ionic.bundle.css" rel="stylesheet">
        <script src="ionic/core/dist/ionic.js"></script>	
        { ... }
    </head>
    <body>
        <ion-tabs>
		  <ion-tab tab="home">Home Content</ion-tab>
		  <ion-tab tab="settings">Settings Content</ion-tab>

		  <ion-tab-bar slot="bottom">

			<ion-tab-button tab="home">
			  <ion-label>Home</ion-label>
			  <ion-icon name="home"></ion-icon>
			  <ion-badge>6</ion-badge>
			</ion-tab-button>

			<ion-tab-button tab="settings">
			  <ion-label>Settings</ion-label>
			  <ion-icon name="settings"></ion-icon>
			</ion-tab-button>

		  </ion-tab-bar>
		</ion-tabs>
        <script type="text/javascript" src="cordova.js"></script>
        <script type="text/javascript" src="js/index.js"></script>
    </body>
</html>
```

**Other information:**
* `androidFetchWorkaround` has been applied to allow for `fetch:///`
* `package.json` dependencies:
```
"dependencies": {
        "@ionic/core": "^4.0.0-rc.2",
        "cordova-android": "^7.1.4",
        "cordova-plugin-ionic-webview": "^2.1.0",
        "cordova-plugin-whitelist": "^1.3.3"
    }
```
* Any fixes suggested in [#14509](https://github.com/ionic-team/ionic/issues/14509) have been tried 

**Ionic info:** 

Note: cli was not used to create the project, `cordova create` was used. Using the CLI to generate a demo app works fine since they seem to use the angular version of Ionic-v4
```
[WARN] You are not in an Ionic project directory. Project context may be 
       missing.

Ionic:

   ionic (Ionic CLI) : 4.3.1

System:

   NodeJS : v8.11.3
   npm    : 6.4.1
   OS     : Linux 4.15
```

**Comment by paulstelzer** [comment]
Thanks for the issue! Do not use the fixes in #14509. It has been changed a lot since than ;)

You are creating the app with pure js, no framework, right?

I cannot say if ionicons will installed if you only install @ionic/core. Take a look if you have node_modules/ionicons. If not, install ionicons.

Go into` node_modules/ionicons/dist/ionicons/svg` and copy everything into yourapp/svg

Then it should work. I don't know what's the correct path is. If you are updating to ionic cordova webview 3, you will see what files are missing (404 errors will come up in console).

**Comment by wcmhc** [comment]
Yes, the app is pure JS and uses just `@ionic/core` to make use of the wonderful Web Components. 

Using the WebView3 plugin was a suggestion from #14509 iirc and is not one of our requirements. 
Ionicons is indeed a dependency of `@ionic/core`. Copying `node_modules/ionicons/dist/ionicons/svg ` -> `www/svg` did not solve the issue. There were never any errors, or any other information as to what the error might be.

An image of Chrome Dev Tools with a screencast of the physical device + network + console: https://i.imgur.com/YZEjUhZ.jpg

The `window.location.href = "";` is used to provide a full view of what the network is doing.

**Comment by paulstelzer** [comment]
Any idea to fix it? I am not using pure JS, so I cannot help you here further

**Comment by wcmhc** [comment]
I have none, the only other mention I can find is from @TomDemulierChevret on the #14509 issue: https://github.com/ionic-team/ionic/issues/14509#issuecomment-400228777

**Comment by paulstelzer** [comment]
Which Android version do you use? emulator or real device?

**Comment by wcmhc** [comment]
Both:
* Emulator: Works correctly and is on Android 5.1.1 (my bad on the original issue being wrong in this regard)
* Physical/Real device: Icons do not show/div.icon-inner is not being populated is on Android 6.0

**Comment by ffMathy** [comment]
Same for me.

**Comment by peterennis** [comment]
Same here.
This repo: https://aetabs.adaept.com
All icons and menu burger are missing in the PWA.
![capture480](https://user-images.githubusercontent.com/140737/51225304-d2dd7780-18fe-11e9-9c86-352beb5e6d4b.png)
![capture479](https://user-images.githubusercontent.com/140737/51225305-d2dd7780-18fe-11e9-9715-122b8607cb12.png)
![capture478](https://user-images.githubusercontent.com/140737/51225306-d3760e00-18fe-11e9-89ef-1ddde48a854f.png)

**Comment by paulstelzer** [comment]
This issue topic was about using ionicons without Angular. @ffMathy @peterennis Using ionicons with `@ionic/angular` is working on PWA, native Android, native iOS wthout problems (of course not in Android 4 and all out-of-date Android Webviews). For more information, please take a look at https://github.com/ionic-team/ionic/issues/17019#issuecomment-452828443

**Comment by ffMathy** [comment]
I found the needed fix. For me, the icon dom was emitted, but the styles were not included. I included them in my global.scss file from the ionicons dist folder with an import.

This feels hacky though. The svgs are properly copied to the wwwroot folder, so my angular.json works.

But the styling isn't working. 

That seems a bit hackish, and feels wrong. Can you confirm this is the way to fix it? 

I should mention that this was a rc0 project that was upgraded to rc1, but it was created in an early beta. Perhaps that's the issue?

**Comment by paulstelzer** [comment]
@ffMathy How does your global.scss look like? In general this is needed:

https://github.com/ionic-team/starters/blob/master/angular/base/src/global.scss

I added this directly to the angular.json:

```
 "styles": [
              {
                "input": "node_modules/@ionic/core/css/ionic.bundle.css"
              },
]
```

**Comment by ffMathy** [comment]
Aha, I didn't know that was possible. Will try that when I get home.

**Comment by wcmhc** [comment]
Updated the test repo to 4.0.0-rc.2 and issue is present
For testing, I allowed all source locations for resources in the Cordova app and loaded Ionic/Core from the unpkg locations. Same result.

**Comment by wcmhc** [comment]
I've solved this after some investigation into the Ionicon repo: The "standard" fetch replacement that's passed around to fix Android's problem with not being able to load SVG's over file:/// doesn't include a `status` property in the returned object, only "ok" and "text". Adding this property allows all icons to load since it no longer fails the `isStatusValid()` check. Thanks for the help @paulstelzer :)

TL;DR: Problem is not in Ionic, use an `androidFetchWorkaround` with correct properties
